src/overlays/actors/ovl_En_Water_Effect/z_en_water_effect.c

```diff
@@ -28,30 +33,672 @@ const ActorInit En_Water_Effect_InitVars = {
+                            EffectSsGRipple_Spawn(globalCtx, &ptr->unk_04, 70, 500, 0);
+                            EffectSsGRipple_Spawn(globalCtx, &ptr->unk_04, 70, 500, 10);
+                            Math_Vec3f_Copy(&sp98, &ptr->unk_04);
+                            sp98.y += 10.0f;
+                            EffectSsGSplash_Spawn(globalCtx, &sp98, NULL, NULL, 1, 300);
+                        }
+                    }
+
+                    for (j = 0; j < 12; j++) {
+                        Matrix_InsertYRotation_f((2.0f * (j * M_PI)) / 5.5f, MTXMODE_NEW);
+                        Matrix_GetStateTranslationAndScaledZ(Rand_ZeroFloat(1.5f) + 1.5f, &spA4);
+                        spA4.y = Rand_ZeroFloat(4.0f) + 2.0f;
+                        func_80A58908(this, &ptr->unk_04, &spA4, ptr->unk_2A);
+                    }
+                }
+            } else if (ptr->unk_00 == 2) {
+                ptr->unk_2C.z += 0.15f;
+                if (ptr->unk_10.y < -8.0f) {
+                    ptr->unk_10.y = -8.0f;
+                }
+
+                if (ptr->unk_10.y < 0.0f) {
+                    if (ptr->unk_04.y <= this->actor.floorHeight) {
+                        ptr->unk_04.y = this->actor.floorHeight;
+                        if (ptr->unk_2A == 0) {
+                            ptr->unk_00 = 3;
+                            ptr->unk_2C.x = 0.05f;
+                            ptr->unk_2C.y = 0.2f;
+                            ptr->unk_10 = D_80A5AFB0;
+                            ptr->unk_1C = D_80A5AFB0;
+                            ptr->unk_3C = 150;
+                            ptr->unk_28 = Rand_ZeroFloat(5.0f) + 7.0f;
+                        } else {
+                            EffectSsGRipple_Spawn(globalCtx, &ptr->unk_04, 0, 200, 0);
+                            ptr->unk_00 = 0;
+                        }
+                    }
+                }
+            } else if (ptr->unk_00 == 3) {
+                Math_ApproachF(&ptr->unk_2C.x, ptr->unk_2C.y, 0.1f, 0.6f);
+                ptr->unk_3C -= ptr->unk_28;
+                if (ptr->unk_3C <= 0) {
+                    ptr->unk_3C = 0;
+                    ptr->unk_00 = 0;
+                }
+            }
+        }
+    }
+}
+
+void EnWaterEffect_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    EnWaterEffect* this = THIS;
+    s32 pad;
+    EnWaterEffectStruct* backupPtr = &this->unk_144[0];
+    EnWaterEffectStruct* ptr = backupPtr;
+    u8 phi_s4 = false;
+    s16 i;
+
+    OPEN_DISPS(gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_144) / 2; i++, ptr++) {
+        if ((ptr->unk_00 == 1) || (ptr->unk_00 == 2)) {
+            if (!phi_s4) {
+                POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 0);
+
+                gSPSegment(POLY_XLU_DISP++, 0x08, Lib_SegmentedToVirtual(D_0408DBE0));
```

Comment:
`D_0408DBE0` is from gameplay keep. Should be able to include and use the symbol.